MetadataExtractor/Formats/Tga/TgaHeaderReader.cs

```diff
@@ -0,0 +1,272 @@
+﻿// Copyright (c) Drew Noakes and contributors. All Rights Reserved. Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+
+using MetadataExtractor.IO;
+using System;
+using System.IO;
+
+namespace MetadataExtractor.Formats.Tga
+{
+    /// <summary>Reads TGA image file header.</summary>
+    /// <author>Dmitry Shechtman</author>
+    sealed class TgaHeaderReader : TgaDirectoryReader<TgaHeaderDirectory, IndexedReader>
+    {
+        struct ColormapInfo
+        {
+            public byte type;
+            public int origin;
+            public int length;
+            public int depth;
+        }
+
+        public const int HeaderSize = 18;
+
+        public static readonly TgaHeaderReader Instance = new TgaHeaderReader();
+
+        private TgaHeaderReader()
+        {
+        }
+
+        public bool TryExtract(byte[] buffer, out TgaHeaderDirectory directory)
```

Comment:
Please set `directory` to `null` when returning `false`. Consuming code might one day reasonably expect or depend upon that.